specification/loadtestservice/LoadTestService/models.tsp

```diff
@@ -62,81 +79,128 @@ enum FileStatus {
-  @doc("Test run request is getting cancelled") CANCELLING,
-  @doc("Test run request is cancelled") CANCELLED,
-  @doc("Test run request is failed") FAILED,
-  @doc("Test run JMX file is validated") VALIDATION_SUCCESS,
-  @doc("Test run JMX file validation is failed") VALIDATION_FAILURE,
+  @doc("Test run is getting provision.") PROVISIONING,
+  @doc("Test run is provisioned.") PROVISIONED,
+  @doc("Test run is getting configure.") CONFIGURING,
+  @doc("Test run configuration is done.") CONFIGURED,
+  @doc("Test run has started executing.") EXECUTING,
+  @doc("Test run has been executed.") EXECUTED,
+  @doc("Test run is getting deprovision.") DEPROVISIONING,
+  @doc("Test run request is deprovisioned.") DEPROVISIONED,
+  @doc("Test run request is finished.") DONE,
+  @doc("Test run request is getting cancelled.") CANCELLING,
+  @doc("Test run request is cancelled.") CANCELLED,
+  @doc("Test run request is failed.") FAILED,
+  @doc("Test run JMX file is validated.") VALIDATION_SUCCESS,
+  @doc("Test run JMX file validation is failed.") VALIDATION_FAILURE,
 }
 
+@doc("Aggregation type.")
 enum AggregationType {
-  @doc("Average value") Average,
-  @doc("Total count") Count,
-  @doc("Aggregation will be average in this case") None,
-  @doc("Total sum") Total,
-  @doc("90th percentile") Percentile90,
-  @doc("95th percentile") Percentile95,
-  @doc("99th percentile") Percentile99,
+  @doc("Average value.") Average,
+  @doc("Total count.") Count,
+  @doc("Aggregation will be average in this case.") None,
+  @doc("Total sum.") Total,
+  @doc("90th percentile.") Percentile90,
+  @doc("95th percentile.") Percentile95,
+  @doc("99th percentile.") Percentile99,
 }
 
+@doc("Metric unit.")
 enum MetricUnit {
-  @doc("No unit specified") NotSpecified,
-  @doc("Percentage") Percent,
-  @doc("Value count") Count,
-  @doc("Seconds") Seconds,
+  @doc("No unit specified.") NotSpecified,
+  @doc("Percentage.") Percent,
+  @doc("Value count.") Count,
+  @doc("Seconds.") Seconds,
   @doc("Milliseconds") Milliseconds,
   @doc("Bytes") Bytes,
   @doc("Bytes per second") BytesPerSecond,
   @doc("Count per second") CountPerSecond,
 }
 
+@doc("Time Grain")
 enum TimeGrain {
-  @doc("5 seconds, available only if test run duration is less than 10 minutes")
+  @doc("5 seconds, available only if test run duration is less than 10 minutes.")
   PT5S,
-  @doc("10 seconds, available only if test run duration is less than 10 minutes")
+
+  @doc("10 seconds, available only if test run duration is less than 10 minutes.")
   PT10S,
   @doc("1 minute") PT1M,
-  @doc("5 minutes, available only if test run duration is greater than 1 minute")
+
+  @doc("5 minutes, available only if test run duration is greater than 1 minute.")
   PT5M,
-  @doc("1 hour, available only if test run duration is greater than 1 minute")
+
+  @doc("1 hour, available only if test run duration is greater than 1 minute.")
   PT1H,
 }
 
+@doc("Intervals")
 enum Interval {
```

Comment:
whats the difference between TimeGrain and Interval enums? do we need two different models?

Reply:
No Pratibha both are same with only difference being the description. Will make this one model and update description accordingly.